packages/salesforcedx-vscode-apex-debugger/src/core/streamingClient.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Client as FayeClient } from 'faye';
+import os = require('os');
+import { DEFAULT_STREAMING_TIMEOUT } from '../constants';
+import { nls } from '../messages';
+
+export enum ApexDebuggerEventType {
+  ApexException,
+  Debug,
+  HeartBeat,
+  LogLine,
+  OrgChange,
+  Ready,
+  RequestStarted,
+  RequestFinished,
+  Resumed,
+  SessionTerminated,
+  Stopped,
+  SystemInfo,
+  SystemGack,
+  SystemWarning
+}
+
+export interface StreamingEvent {
+  createdDate: string;
+  replayId: number;
+  type: string;
+}
+
+export interface ApexDebuggerEvent {
+  SessionId: string;
+  RequestId?: string;
+  BreakpointId?: string;
+  Type: string;
+  Description?: string;
+  FileName?: string;
+  Line?: number;
+  Stacktrace?: string;
+}
+
+export interface DebuggerMessage {
+  event: StreamingEvent;
+  sobject: ApexDebuggerEvent;
+}
+
+export class StreamingClientInfo {
+  public channel: string;
+  public timeout: number;
+  public errorHandler: Function;
```

Comment:
If `Function` a built-in type for TypeScript? What is its signature like? any => any?